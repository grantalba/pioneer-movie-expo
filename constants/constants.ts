export type OnboardingScreensType = {
  id: number;
  image: any;
  title: string;
  desc: string;
};

export type MovieDetailsType = {
  poster_path: string;
  backdrop_path: string;
  title: string;
  overview: string;
  vote_average: string;
  id: number;
};

export const onboarding_screens: OnboardingScreensType[] = [
  {
    id: 1,
    image: require('@/assets/images/splash_1.png'),
    title: 'Watch Movies',
    desc: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. In dapibus varius nulla, quis dictum augue.',
  },
  {
    id: 2,
    image: require('@/assets/images/splash_2.png'),
    title: 'Search Movies',
    desc: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. In dapibus varius nulla, quis dictum augue.',
  },
  {
    id: 3,
    image: require('@/assets/images/splash_3.png'),
    title: 'Download Movies',
    desc: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. In dapibus varius nulla, quis dictum augue.',
  },
];
