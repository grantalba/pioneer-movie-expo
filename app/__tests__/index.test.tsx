import { NavigationContainer } from '@react-navigation/native';
import { render, fireEvent, screen, RenderAPI } from '@testing-library/react-native';
import { useRouter } from 'expo-router';
import React from 'react';

import { onboarding_screens } from '@/constants/constants';

import Index from '../index';

jest.mock('expo-router', () => ({
  useRouter: jest.fn(),
}));

// Mock react-native-safe-area-context
jest.mock('react-native-safe-area-context', () => {
  return {
    useSafeAreaInsets: () => ({
      top: 0,
      bottom: 0,
      left: 0,
      right: 0,
    }),
  };
});

const mockNavigate = jest.fn();
beforeEach(() => {
  (useRouter as jest.Mock).mockReturnValue({ navigate: mockNavigate });
});

// Render the component with providers
const renderWithProviders = () =>
  render(
    <NavigationContainer>
      <Index />
    </NavigationContainer>
  );

const allDescElements = (screen: RenderAPI) => {
  // Get all kinds of Lorem Ipsum Content
  return screen.getAllByText(onboarding_screens[0].desc);
};

describe('Index (Onboarding) Screen', () => {
  it('renders first onboarding screen', async () => {
    renderWithProviders();

    expect(screen.getByText(onboarding_screens[0].title)).toBeOnTheScreen();
    expect(allDescElements(screen)[0]).toHaveTextContent(onboarding_screens[0].desc);
    expect(screen.getByText('Next')).toBeOnTheScreen();
  });

  it('renders second onboarding screen', () => {
    renderWithProviders();

    const button = screen.getByLabelText('onboardingTextButton');

    fireEvent.press(button);

    expect(screen.getByText(onboarding_screens[1].title)).toBeOnTheScreen();
    expect(allDescElements(screen)[1]).toHaveTextContent(onboarding_screens[1].desc);
    expect(screen.getByText('Next')).toBeOnTheScreen();
  });

  it('render third onboarding screen and navigates to homescreen on last press', async () => {
    renderWithProviders();
    const button = screen.getByLabelText('onboardingTextButton');

    for (let i = 0; i < onboarding_screens.length; i++) {
      fireEvent.press(button);

      if (i === onboarding_screens.length) {
        expect(screen.getByText(onboarding_screens[2].title)).toBeOnTheScreen();
        expect(allDescElements(screen)[2]).toHaveTextContent(onboarding_screens[2].desc);
        expect(screen.getByText(`Let's Go`)).toBeOnTheScreen();
      }
    }

    expect(mockNavigate).toHaveBeenCalledWith('/homescreen');
  });
});
