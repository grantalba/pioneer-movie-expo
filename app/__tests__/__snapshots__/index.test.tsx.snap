// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index (Onboarding) Screen should match the snapsho 1`] = `
<View
  style={
    {
      "alignItems": "stretch",
      "backgroundColor": "#212122",
      "flex": 1,
      "flexDirection": "column",
      "paddingBottom": 20,
      "paddingHorizontal": 0,
      "paddingTop": 0,
    }
  }
>
  <View
    style={
      {
        "flex": 2,
        "paddingTop": 0,
      }
    }
  >
    <RCTScrollView
      data={
        [
          {
            "desc": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. In dapibus varius nulla, quis dictum augue.",
            "id": 1,
            "image": {
              "testUri": "../../../assets/images/splash_1.png",
            },
            "title": "Watch Movies",
          },
          {
            "desc": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. In dapibus varius nulla, quis dictum augue.",
            "id": 2,
            "image": {
              "testUri": "../../../assets/images/splash_2.png",
            },
            "title": "Search Movies",
          },
          {
            "desc": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. In dapibus varius nulla, quis dictum augue.",
            "id": 3,
            "image": {
              "testUri": "../../../assets/images/splash_3.png",
            },
            "title": "Download Movies",
          },
        ]
      }
      decelerationRate="fast"
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={true}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      pagingEnabled={true}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEnabled={false}
      scrollEventThrottle={0.0001}
      showsHorizontalScrollIndicator={false}
      snapToAlignment="center"
      snapToInterval={750}
      stickyHeaderIndices={[]}
      viewabilityConfigCallbackPairs={[]}
    >
      <View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={
            [
              {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "justifyContent": "center",
                "width": 750,
              }
            }
          >
            <Image
              resizeMode="contain"
              source={
                {
                  "testUri": "../../../assets/images/splash_1.png",
                }
              }
              style={
                {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": 667,
                  "justifyContent": "center",
                  "width": 600,
                }
              }
            />
          </View>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={
            [
              {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "justifyContent": "center",
                "width": 750,
              }
            }
          >
            <Image
              resizeMode="contain"
              source={
                {
                  "testUri": "../../../assets/images/splash_2.png",
                }
              }
              style={
                {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": 667,
                  "justifyContent": "center",
                  "width": 600,
                }
              }
            />
          </View>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={
            [
              {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "justifyContent": "center",
                "width": 750,
              }
            }
          >
            <Image
              resizeMode="contain"
              source={
                {
                  "testUri": "../../../assets/images/splash_3.png",
                }
              }
              style={
                {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": 667,
                  "justifyContent": "center",
                  "width": 600,
                }
              }
            />
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
  <View
    style={
      {
        "flex": 1,
        "justifyContent": "space-around",
      }
    }
  >
    <RCTScrollView
      data={
        [
          {
            "desc": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. In dapibus varius nulla, quis dictum augue.",
            "id": 1,
            "image": {
              "testUri": "../../../assets/images/splash_1.png",
            },
            "title": "Watch Movies",
          },
          {
            "desc": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. In dapibus varius nulla, quis dictum augue.",
            "id": 2,
            "image": {
              "testUri": "../../../assets/images/splash_2.png",
            },
            "title": "Search Movies",
          },
          {
            "desc": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. In dapibus varius nulla, quis dictum augue.",
            "id": 3,
            "image": {
              "testUri": "../../../assets/images/splash_3.png",
            },
            "title": "Download Movies",
          },
        ]
      }
      decelerationRate="fast"
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={true}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      pagingEnabled={true}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEnabled={false}
      scrollEventThrottle={0.0001}
      showsHorizontalScrollIndicator={false}
      snapToAlignment="center"
      snapToInterval={750}
      stickyHeaderIndices={[]}
      viewabilityConfigCallbackPairs={[]}
    >
      <View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={
            [
              {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "justifyContent": "center",
                "width": 750,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#D4D4F6",
                  "fontSize": 32,
                  "fontWeight": "600",
                  "lineHeight": 40,
                  "textAlign": "center",
                }
              }
            >
              Watch Movies
            </Text>
            <Text
              style={
                {
                  "color": "#D4D4F6",
                  "fontSize": 18,
                  "fontWeight": "300",
                  "lineHeight": 24,
                  "marginHorizontal": 20,
                  "marginTop": 12,
                  "textAlign": "center",
                }
              }
            >
              Lorem ipsum dolor sit amet, consectetur adipiscing elit. In dapibus varius nulla, quis dictum augue.
            </Text>
          </View>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={
            [
              {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "justifyContent": "center",
                "width": 750,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#D4D4F6",
                  "fontSize": 32,
                  "fontWeight": "600",
                  "lineHeight": 40,
                  "textAlign": "center",
                }
              }
            >
              Search Movies
            </Text>
            <Text
              style={
                {
                  "color": "#D4D4F6",
                  "fontSize": 18,
                  "fontWeight": "300",
                  "lineHeight": 24,
                  "marginHorizontal": 20,
                  "marginTop": 12,
                  "textAlign": "center",
                }
              }
            >
              Lorem ipsum dolor sit amet, consectetur adipiscing elit. In dapibus varius nulla, quis dictum augue.
            </Text>
          </View>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={
            [
              {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "justifyContent": "center",
                "width": 750,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#D4D4F6",
                  "fontSize": 32,
                  "fontWeight": "600",
                  "lineHeight": 40,
                  "textAlign": "center",
                }
              }
            >
              Download Movies
            </Text>
            <Text
              style={
                {
                  "color": "#D4D4F6",
                  "fontSize": 18,
                  "fontWeight": "300",
                  "lineHeight": 24,
                  "marginHorizontal": 20,
                  "marginTop": 12,
                  "textAlign": "center",
                }
              }
            >
              Lorem ipsum dolor sit amet, consectetur adipiscing elit. In dapibus varius nulla, quis dictum augue.
            </Text>
          </View>
        </View>
      </View>
    </RCTScrollView>
    <View
      accessibilityLabel="onboardingTextButton"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#706FD3",
          "borderRadius": 22,
          "height": 55,
          "justifyContent": "center",
          "marginHorizontal": 24,
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          {
            "color": "#EEEEEE",
            "fontSize": 18,
            "lineHeight": 24,
          }
        }
      >
        Next
      </Text>
    </View>
  </View>
</View>
`;
